ui/lib/CreateStream.tsx

```diff
@@ -25,32 +26,67 @@ export interface CreateStreamProps {
 }
 
 const CreateStream = (props: CreateStreamProps) => {
+  const [balance, setBalance] = useState<number | null>(null);
+  const [tokenBalance, setTokenBalance] = useState<number | null>(null);
+  const [gasPrice, setGasPrice] = useState<number | null>(null);
+  const { data, isError, isLoading } = useFeeData();
   const { address } = useAccount();
+
+  // Set balances for native and stream tokens
+  useEffect(() => {
+    if (address) {
+      (async () => {
+        // Fetch and set native balance as integer
+        const balanceData = await fetchBalance({ address });
+        if (balanceData && balanceData.formatted !== undefined) {
+          setBalance(Number(balanceData.formatted));
+        }
+        // Fetch and set token balance as integer
+        const tokenBalanceData = await fetchBalance({
```

Comment:
Can this one be performed in parallel to the above by wrapping in a 2nd `useEffect`?

Reply:
Very good idea; separation of concerns 101; updated